frappe/query_builder/qb.py

```diff
@@ -0,0 +1,90 @@
+from pypika import MySQLQuery, Order, PostgreSQLQuery
+from pypika import functions as fn
+from pypika import terms
+from pypika.queries import Schema, Table
+import frappe.query_builder.custom_functions as SpecialFuncs
+
+def qb(db_type):
+	if not db_type:
+		db_type = "mariadb"
+	selecter = {"mariadb": MariaDB, "postgres": Postgres}
+	return selecter[db_type]
+
+class common:
+	fn = fn
+	terms = terms
+	desc = Order.desc
+	Schema = Schema
+
+	@staticmethod
+	def Table(classname:str, *args, **kwargs):
+		if not classname.startswith("__"):
+			classname = "tab" + classname
+		return Table(classname, *args, **kwargs)
+
+class MariaDB(common, MySQLQuery,):
+	Field = terms.Field
+	GROUP_CONCAT = SpecialFuncs.GROUP_CONCAT
+	Match = SpecialFuncs.Match
+
+
+	def __init__(self) -> None:
+		super().__init__()
+
+	@classmethod
+	def from_(cls, class_name, *args, **kwargs):
+		if isinstance(class_name,str):
+			class_name = "tab"+class_name
+		return super().from_(class_name, *args, **kwargs)
+
+	@staticmethod
+	def rename_table(old_name, new_name):
+		return f"RENAME TABLE `tab{old_name}` TO `tab{new_name}`"
+
+	@staticmethod
+	def DESC(dt):
+		return f"DESC `tab{dt}`"
+
+	@staticmethod
+	def change_table_type(tb, col, type):
+		return f"ALTER TABLE `{tb}` MODIFY `{col}` {type} NOT NULL"
+
+
+class Postgres(common, PostgreSQLQuery,):
+	postgres_field = {"table_name": "relname", "table_rows": "n_tup_ins"}
+	information_schema_translation = {"tables": "pg_stat_all_tables"}
```

Comment:
Maintain single mapping for standard pg tables instead

Reply:
`postgres_field` is a mapping of MariaDB columns to Postgres columns and `information_schema_translation` is the mapping for the repective schemas.

The variable names are ambiguous which makes it difficult to understand, but maintaining a single mappting isn't possible assuming there could be different columns names which exist with the same schema name.
 
Maintaining separate dicts is simpler

I've changed the name to  `field_translation` and `schema_translation` so its easier to understand.